service/models.py

```diff
@@ -50,6 +50,18 @@ def create(self):
         self.id = None  # id must be none to generate next primary key
         db.session.add(self)
         db.session.commit()
+        
+     def update(self):
+        """
+        Updates a Recommendation to the database
+        """
+        logger.info("Saving %s", self.name)
+        if not self.id:
+            raise DataValidationError("Update called with empty ID field")
+        if type(self.id) is not int:
+            raise DataValidationError("Update called with non-integer ID field")
+        if type(self.price) is not int:
```

Comment:
There is no price in the Recomendation Model, change the code according to the attributes of Recommendation model i.e src_prod_id etc.